lohica-js-network/src/components/searchPeopleComponents/SearchInputComponent.js

```diff
@@ -0,0 +1,59 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Form, FormControl, ListGroup } from 'react-bootstrap';
+import { searchUsers, emptySearchResult } from '../../actions/searchActions';
+
+class SearchInputComponent extends React.Component{
+    constructor(props){
+        super(props);
+        
+        this.handleChange = this.handleChange.bind(this);
+        this.onKeyPress = this.onKeyPress.bind(this);
+    }
+
+    // also validate the queue with the name regex!!!!
+    handleChange(e) {
+        const { userId } = this.props;
+        const searchQueue = e.target.value;
+        if(searchQueue){
+            this.props.dispatch(searchUsers(searchQueue, userId));
+        } else {
+            this.props.dispatch(emptySearchResult());
+        }
+    }
+
+    // prevent input from submit
+    onKeyPress(e){
+        if(e.which === 13){
```

Comment:
use constants here

Reply:
created constants file, importing button code from it